api/v1/infinispan_types.go

```diff
@@ -633,11 +633,10 @@ const (
 // EncryptionSiteSpec enables TLS for cross-site replication
 type EncryptionSiteSpec struct {
 	// +optional
-	Protocol          TLSProtocol       `json:"protocol,omitempty"`
-	TransportKeyStore CrossSiteKeyStore `json:"transportKeyStore"`
-	RouterKeyStore    CrossSiteKeyStore `json:"routerKeyStore"`
-	// +optional
-	TrustStore *CrossSiteTrustStore `json:"trustStore,omitempty"`
```

Comment:
I don't think we can make this change without breaking CRD compatibility :disappointed:

Reply:
How? It is the same type and name, just not optional.

Reply:
I hope it fails because it isn't secure. See my comment: https://github.com/infinispan/infinispan-operator/pull/1993#issuecomment-1953827521

Reply:
We need to verify this is the case. On upgrade I *think* the webhooks will be invoked for the Infinispan CR. If we verify this is in fact the case and we keep the field as a pointer, then we can potentially return an error to the user informing them what change is required.